etherman/etherman.go

```diff
@@ -245,6 +260,9 @@ func (etherMan *Client) EstimateGasSequenceBatches(sequences []ethmanTypes.Seque
 
 // SequenceBatches send sequences of batches to the ethereum
 func (etherMan *Client) SequenceBatches(sequences []ethmanTypes.Sequence, gasLimit uint64, gasPrice, nonce *big.Int) (*types.Transaction, error) {
+	if etherMan.readOnly {
+		return nil, errors.New("etherman client in read-only mode, cannot send sequences of batches")
+	}
```

Comment:
I'm worried about the sequencer and aggregator behavior when this error is returned

Reply:
@ARR552 I added a test for the specific error you are referring to https://github.com/0xPolygonHermez/zkevm-node/blob/85221c0d861788aef453b8e161ea722bdd590590/ethtxmanager/ethtxmanager_test.go#L23

More broadly, in case of a read-only etherman client, the affected methods are:
- `EstimateGasSequenceBatches`: non-trivial to test. The error is handled by the caller.
- `SequenceBatches`: added UT. The caller handles the error.
- `EstimateGasForVerifyBatch`: method not used.
- `VerifyBatch`: added UT. The caller handles the error.
- `ApproveMatic`: error trivially handled in `cmd/approve.go`
- `GetPublicAddress`: not easy to test. The new error caused the method to change signature. I added the error handling here.
https://github.com/0xPolygonHermez/zkevm-node/blob/85221c0d861788aef453b8e161ea722bdd590590/aggregator/aggregator.go#L367-L370

I don't see major concerns, since the new error is handled in every case. But I clearly don't have the full picture. I trust you to see any blocking point.